### ISSUE TITLE
Implement a JNI binding for EMV-C

### ISSUE THREAD

**Issue by axic** [ethereum/ethereumj#1025]
[EVM-C](http://ethereum.github.io/evmjit/docs/group__EVMC.html) is a thin API to write VMs for and to be used in clients for connecting to VMs.

cpp-ethereum supports it natively to use [evmjit](http://github.com/ethereum/evmjit) and pyethereum/go-ethereum had prototype implementations. More importantly, the upcoming [ewasm](http://github.com/ewasm/design) VM called [Hera](http://github.com/ewasm/hera) also uses this interface.

Supporting it in ethereumj/Harmony would allow a quick way for ethereumj to be part of the ewasm testnet without the need of writing a native Java interpereter for Webassembly.

**Comment by axic** [comment]
Updated, also the repo link, since we've split it out into https://github.com/ethereum/evmc

**Comment by kishansagathiya** [comment]
@zilm13 @mkalinin 
Planning to pick this. Any tips? How big would this be?

**Comment by mkalinin** [comment]
@kishansagathiya do you have an experience in JNI binding development?

**Comment by kishansagathiya** [comment]
@mkalinin Nope. But I can read up stuff(I was few hours back) and try it. Does it seem too difficult to take up?

**Comment by mkalinin** [comment]
@kishansagathiya it's a bit tricky. I would advise you to continue with something else.

**Comment by axic** [comment]
There is some initial work started here: https://github.com/ethereum/evmc/pull/171
